src/gui/dialogs/ColorChooser.cpp

```diff
@@ -0,0 +1,98 @@
+/* ColorChooser.cpp - definition of ColorChooser class.
+ *
+ * Copyright (c) 2020 russiankumar <adityakumar4644/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <ColorChooser.h>
+
+
+
+
+//! Set global palette via array, checking bounds
+void ColorChooser::setPalette (const QVector<QColor> colors)
+{
+	const int max = qMin (colors.size(), 48);
+	for (int i = 0; i < max; i++)
+	{
+		ColorChooser::setStandardColor (i, colors[i]);
+	}
+}
+
+
+//! Set global paletter via enum
+void ColorChooser::setPalette (const Palette palette)
+{
+	switch (palette)
+	{
+		case Palette::Default: setPalette (defaultPalette()); break;
+		case Palette::Mixer: setPalette (nicePalette(140)); break;
+		case Palette::Track: setPalette (nicePalette(150)); break;
+	}
+}
```

Comment:
Now that we have a getPalette, we should use it here to ensure consistency and be DRY.
```suggestion
void ColorChooser::setPalette (const Palette palette)
{
	setPalette(getPalette(palette));
}
```

Reply:
Implemented.